modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -904,9 +953,13 @@ void collectorThread(std::vector<Q*>   in_queues,
         flags[idx] = true;
     }
 
+    GAPI_ITT_STATIC_LOCAL_HANDLE(collector_hndl, "collector_region");
+
     QueueReader qr;
     while (true)
     {
+        GAPI_ITT_AUTO_TRACE_GUARD(collector_hndl);
```

Comment:
This trace seems also extra

Reply:
That's ok to have a "parent" task as the enclosing one

Reply:
Also: the `out_queue.push` thing can be traced, too. What if the whole pipeline gets stuck by the slow user processing code? The `push` time should highlight that.

Reply:
Thanks a lot, reworked!

Reply:
> What if the whole pipeline gets stuck by the slow user processing code? The push time should highlight that.

You can notice it by distance between `GStreamingExecutor::pull` blocks.